engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -320,7 +326,7 @@ public static void main(String[] args) throws Throwable
       logger.error("Fatal {} in container!", (e instanceof Error) ? "Error" : "Exception", e);
       /* Report back any failures, for diagnostic purposes */
       try {
-        umbilical.reportError(childId, null, ExceptionUtils.getStackTrace(e), logFileInfo);
+        umbilical.reportError(childId, null, ExceptionUtils.getStackTrace(e), logFileInfo, DeployManager.generateNewGroupId());
```

Comment:
One suggestion: can we have groupId generation and mapping only on the Stram side? What I was thinking is ideally only the Stram needs to keep track of deploy/undeploy events and that can be done with the DeployManager which now also maintains maps of operatorId -> DeployRequest(s) and ContainerId -> DeployRequest(s). We need to think about whether there can be more than one DeployRequest per operator or container or we should even maintain that.